charts/internal/azure-infra/templates/main.tf

```diff
@@ -73,17 +73,75 @@ resource "azurerm_network_security_group" "workers" {
   {{- end}}
 }
 
+{{ if .Values.create.natGateway -}}
+#===============================================
+#= NAT Gateway
+#===============================================
+{{ if .Values.natGateway -}}
+{{ range $index, $ip := .Values.natGateway.ipAddresses }}
+data "azurerm_public_ip" "nat-ip-{{ $index }}" {
+  name                = "{{ $ip.name }}"
+  resource_group_name = "{{ $ip.resourceGroup }}"
+}
+{{- end }}
+{{ range $index, $ip := .Values.natGateway.ipAddressRanges }}
+data "azurerm_public_ip_prefix" "nat-iprange-prefix-{{ $index }}" {
+  name                = "{{ $ip.name }}"
+  resource_group_name = "{{ $ip.resourceGroup }}"
+}
+{{- end }}
+{{ else -}}
+resource "azurerm_public_ip" "natip" {
+  name                = "{{ required "clusterName is required" .Values.clusterName }}-nat-ip"
+  location            = "{{ required "azure.region is required" .Values.azure.region }}"
+  {{ if .Values.create.resourceGroup -}}
+  resource_group_name = "${azurerm_resource_group.rg.name}"
+  {{- else -}}
+  resource_group_name = "${data.azurerm_resource_group.rg.name}"
+  {{- end }}
+  allocation_method   = "Static"
+  sku                 = "Standard"
+}
+{{- end }}
+
+resource "azurerm_nat_gateway" "nat" {
```

Comment:
cc https://github.com/gardener/gardener-extension-provider-azure/issues/43#issuecomment-598020619 - @vlerenc suggested to have one NAT GW per zone, but it seems that this is not configured here / probably not configurable at all for Azure. Can you shed some light on how Azure handles zone failures for NAT GWs?